Course details components (professors)

lyangji1011 opened the pull request:
### Summary

Resolves #379 

<!-- What does this PR change and why? Discuss any breaking changes. -->
* Creates a `Breadcrumb` component with labels and links
* Creates a `MetricsCard` component that takes in a list of metrics and returns a card of them in a row
* Creates a `ProfessorInfoCard` component that displays prof name and information about sections they teach. Did not include MetricsCard inside this card, since ticket said to ignore fetching data for metrics
* Creates sandbox page at `/sandbox/ProfessorInfo` that displays each component

### Checklist

- [ ] Breadcrumb
- [ ] MetricsCard
- [ ] ProfessorInfoCard


### How to Test
<!-- Describe how to test your code. -->
Testing sandbox page at `/sandbox/ProfessorInfo`
* Displays Breadcrumb and MetricsCard using hardcoded dummy data
* Uses ScheduleContext to get Course Object, currently is hardcoded to Frederic Faulkner courses. 
  * This also involved changing the App component, since we want to use the same context as App for AppDataLoader (to load information about section seats)
<img width="1500" height="481" alt="Screenshot 2025-11-04 at 1 44 31 AM" src="https://github.com/user-attachments/assets/0ff111e5-73ef-4628-9584-a9575d129312" />
<img width="3000" height="920" alt="image" src="https://github.com/user-attachments/assets/63505329-c221-4bc6-87b3-3bac8b289ff1" />

DanielChee committed:
Daniel/superior seat display mechanism (#359)

### Summary

Resolves #353 

<!-- What does this PR change and why? Discuss any breaking changes. -->

caches seating info and removes tooltip from existing design.
conditionally renders seat and waitlist info based on color of
background.

### How to Test

flame my code
but my functionality is flawless
bog

---------

Co-authored-by: Uma Anand <63426041+uma-anand@users.noreply.github.com>
Co-authored-by: Uma Anand <uma2005anand@gmail.com>

afazio1 committed:
Fix seating info style inconsistencies (#361)

### Summary

- wrong font color
- incorrect padding




### How to Test
Deploy preview

---------

Co-authored-by: Uma Anand <uma2005anand@gmail.com>

lyangji1011 committed:
in progress breadcrumb

lyangji1011 committed:
metrics card

lyangji1011 committed:
complete info card

lyangji1011 committed:
linting

lyangji1011 committed:
fixes

lyangji1011 committed:
light/dark mode fixes

uma-anand commented on src/components/Breadcrumb/stylesheet.scss:
```diff
@@ -0,0 +1,34 @@
+      align-items: center;
+
+      .link {
+        color: #8bd6fb;
```
Can you add this to the variables.scss file if it isnt already there and replace this with a reference to that?

uma-anand commented on src/components/Breadcrumb/stylesheet.scss:
```diff
@@ -0,0 +1,34 @@
+  align-items: center;
+  font-size: 16px;
+  padding: 8px 12px;
+  font-family: 'Roboto';
```
Is this font-family really necessary? Isn't this the default font? If this is the case, just change it everywhere this occurs

uma-anand commented on src/components/Breadcrumb/stylesheet.scss:
```diff
@@ -0,0 +1,34 @@
+.Breadcrumb {
+  display: flex;
+  align-items: center;
+  font-size: 16px;
```
The codebase tries to use 'em' instead of px for fontsize. Can you change the px to em in all places this occurs?

uma-anand commented on src/components/ProfessorInfoCard/index.tsx:
```diff
@@ -0,0 +1,172 @@
+            <FontAwesomeIcon
+              icon={isPinned ? faCheck : faPlus}
+              style={{
+                color: '#8BD6FB',
```
add to variables if not already there

uma-anand commented on src/components/ProfessorInfoCard/index.tsx:
```diff
@@ -0,0 +1,172 @@
+  const [{ pinnedCrns }, { patchSchedule }] = useContext(ScheduleContext);
+
+  const handleAddSection = (section: Section): void => {
+    patchSchedule({ pinnedCrns: [...pinnedCrns, section.crn] });
```
While this adds it to the schedule, I don't see it being added on the left side panel.

uma-anand commented on src/components/ProfessorInfoCard/index.tsx:
```diff
@@ -0,0 +1,172 @@
+  );
+}
+
+function formatTime(minutes: number): string {
```
this should be in utils/misc.tsx

uma-anand commented on src/components/ProfessorInfoCard/index.tsx:
```diff
@@ -0,0 +1,172 @@
+  waitlist: OccupiedInfo | null;
+};
+
+const fetchSeating = async (
```
There is a lot of overlap between this and Daniel's code. Can you try creating a value returning SeatData in the section bean instead of doing this each time? Then you can patch yours and Daniels code to just use that instead.

uma-anand commented on src/components/ProfessorInfoCard/stylesheet.scss:
```diff
@@ -0,0 +1,106 @@
+  }
+
+  .sections-container {
+    border-radius: 8px;
```
why is a border radius needed here?

uma-anand commented on src/components/ProfessorInfoCard/stylesheet.scss:
```diff
@@ -0,0 +1,106 @@
+        display: flex;
+        align-items: center;
+
+        &:nth-child(1) { width: 60px; margin-right: 20px; }
```
confused about why we are using hardcoded width and margins. The widths are fine except for the last column. Also try to replace them with percentages. I notice paddings in the Figma- is there a reason you used margins?

uma-anand commented on src/components/ProfessorInfoCard/stylesheet.scss:
```diff
@@ -0,0 +1,106 @@
+            justify-content: center;
+          }
+
+          &:nth-child(1) { width: 60px; margin-right: 20px; }
```
same note as before. Also if this is repeating, try to combine with above.

uma-anand commented on src/components/SeatInfo/index.tsx:
Can you uninclude the changes from other PRs?

uma-anand reviewed (changes requested):
Code is clean! Thanks for implementing the components in the specified way. I've requested a couple changes.

Also, curious about the light mode design, did you talk to Cynthia about it? Otherwise, you may want to check with her.

uma-anand commented:
Also, one thing @afazio1 pointed out is while we don't have mobile designs just yet, we still want to make this adaptive to other screen sizes. Right now, it doesn't resize even for smaller desktops. See if you can make it a little more responsive.

lyangji1011 committed:
unselected courses added as desired

lyangji1011 committed:
Merge remote-tracking branch 'origin/bog-f25-sprint4' into lauren/course-details-profs

lyangji1011 committed:
function for seat data

lyangji1011 committed:
responsive rows + clean up css

lyangji1011 committed:
fix

lyangji1011 commented:
I pretty much addressed everything:
- Cynthia approved of the light mode design
- I used hardcoded widths because they need to be aligned between all the rows, including the header. Not sure if there's a way to do it without hardcoding? I did make it more responsive though, and switched to percentages
- Also switched to using padding between the row cells. Sorry, did not notice it was padding in Figma
- Combined styling for header row and section rows, so the styling is not redundant anymore

uma-anand reviewed (commented):
Looks good! You can

1. Go ahead and remove the sandbox now
2. Just make the metric card float to next line instead of cropping out for smaller screen sizes

And then you should be good to go!!

uma-anand commented:
Mobile designs are now on the Figma for you to take a look at. Let me know if you have time to implement, otherwise I can take a look.

lyangji1011 committed:
responsive metrics card

lyangji1011 committed:
remove sandbox page

uma-anand reviewed (approved):
LGTM! Thanks for working through all the changes!
